src/main/java/com/cloudbees/jenkins/support/slowrequest/SlowRequestThreadDumpsGenerator.java

```diff
@@ -0,0 +1,121 @@
+
+    /**
+     * The minimal number of SlowRequest found at the same time (in the last 3 seconds) to trigger the ThreadDump generation
+     */
+    public static final int MINIMAL_SLOW_REQUEST_COUNT =
+            Integer.getInteger(SlowRequestThreadDumpsGenerator.class.getName() + ".MINIMAL_SLOW_REQUEST_COUNT", 5);
+
+    /**
+     * Number of ThreadDump that will be generated during the slowRequest scenario
+     */
+    public static final int TOTAL_ITERATIONS =
+            Integer.getInteger(SlowRequestThreadDumpsGenerator.class.getName() + ".TOTAL_ITERATIONS", 4);
+
+    /**
+     * Time in seconds that we will wait between the ThreadDump generations (under the same slowRequest check)
+     */
+    public static final int FREQUENCY_SEC =
+            Integer.getInteger(SlowRequestThreadDumpsGenerator.class.getName() + ".FREQUENCY_SEC", 5);
+
+    /**
+     * Limit the number of thread dumps to retain on slowRequest scenario
+     */
+    public static final int SLOW_REQUEST_THREAD_DUMPS_TO_RETAIN = Integer.getInteger(
+            SlowRequestThreadDumpsGenerator.class.getName() + ".SLOW_REQUEST_THREAD_DUMPS_TO_RETAIN", 40);
+
+    /**
+     * Thread dumps generated on slowRequest scenario are stored in $JENKINS_HOME/slow-request-threaddumps
+     */
+    protected final FileListCap logs = new FileListCap(
+            new File(Jenkins.get().getRootDir(), "slow-request-threaddumps"), SLOW_REQUEST_THREAD_DUMPS_TO_RETAIN);
+
+    /**
+     * Provide a means to disable the slow request thread dump checker. This is a volatile non-final field as if you run into
+     * issues in a running Jenkins you may need to disable without restarting Jenkins.
+     */
+    public static volatile boolean DISABLED =
+            Boolean.getBoolean(SlowRequestThreadDumpsGenerator.class.getName() + ".DISABLED");
+
+    private final SimpleDateFormat format = new SimpleDateFormat("yyyyMMdd-HHmmss.SSS");
+
+    private long iota = 0l;
+
+    public SlowRequestThreadDumpsGenerator(long iota) {
+        this.iota = iota;
+    }
+
+    public SlowRequestThreadDumpsGenerator() {}
+
+    @Override
+    public void run() {
+        if (DISABLED) {
+            return;
+        }
+
+        long fileNameDate = this.iota;
+
+        for (int i = 0; i < TOTAL_ITERATIONS; i++) {
+            File threadDumpFile = logs.file(format.format(new Date(fileNameDate)) + ".txt");
+            try (FileOutputStream fileOutputStream = new FileOutputStream(threadDumpFile)) {
+                ThreadDumps.threadDump(fileOutputStream);
+                logs.add(threadDumpFile);
+            } catch (IOException ioe) {
+                LOGGER.log(
+                        WARNING,
+                        "Support Core plugin can't generate automatically thread dumps under SlowRequest scenario",
+                        ioe);
+            } finally {
+                fileNameDate += FREQUENCY_SEC * 1000;
+            }
+        }
```

Comment:
Hmm maybe I am missing something here, but we never actually sleep or reschedule anything here, so we are actually generating `TOTAL_ITERATIONS` thread dumps with different file names in a loop, not every `FREQUENCY_SEC * 1000` seconds ?

Reply:
Yeah definitely generating dump without the 5 sec interval:

```
% ll /tmp/je-1-home/slow-request-threaddumps
total 436
drwxr-xr-x  6 jenkins wheel    192 Jul 19 19:58 .
drwxr-xr-x 53 jenkins wheel   1696 Jul 19 19:56 ..
-rw-r--r--  1 jenkins wheel 120211 Jul 19 19:58 20240719-195806.303.txt
-rw-r--r--  1 jenkins wheel 105633 Jul 19 19:58 20240719-195811.303.txt
-rw-r--r--  1 jenkins wheel 106723 Jul 19 19:58 20240719-195816.303.txt
-rw-r--r--  1 jenkins wheel 105673 Jul 19 19:58 20240719-195821.303.txt
% stat /tmp/je-1-home/slow-request-threaddumps/20240719-195806.303.txt
16777233 9837190 -rw-r--r-- 1 jenkins wheel 0 120211 "Jul 19 19:58:06 2024" "Jul 19 19:58:06 2024" "Jul 19 19:58:06 2024" "Jul 19 19:58:06 2024" 4096 240 0 /tmp/je-1-home/slow-request-threaddumps/20240719-195806.303.txt
% stat /tmp/je-1-home/slow-request-threaddumps/20240719-195811.303.txt
16777233 9837191 -rw-r--r-- 1 jenkins wheel 0 105633 "Jul 19 19:58:07 2024" "Jul 19 19:58:06 2024" "Jul 19 19:58:06 2024" "Jul 19 19:58:06 2024" 4096 208 0 /tmp/je-1-home/slow-request-threaddumps/20240719-195811.303.txt
% stat /tmp/je-1-home/slow-request-threaddumps/20240719-195816.303.txt
16777233 9837194 -rw-r--r-- 1 jenkins wheel 0 106723 "Jul 19 19:58:07 2024" "Jul 19 19:58:06 2024" "Jul 19 19:58:06 2024" "Jul 19 19:58:06 2024" 4096 216 0 /tmp/je-1-home/slow-request-threaddumps/20240719-195816.303.txt
% stat /tmp/je-1-home/slow-request-threaddumps/20240719-195821.303.txt
16777233 9837195 -rw-r--r-- 1 jenkins wheel 0 105673 "Jul 19 19:58:08 2024" "Jul 19 19:58:07 2024" "Jul 19 19:58:07 2024" "Jul 19 19:58:06 2024" 4096 208 0 /tmp/je-1-home/slow-request-threaddumps/20240719-195821.303.txt
```

Reply:
Holy sh.. You are right. I forgot to add the sleeping call to wait for this `FREQUENCY_SEC * 1000` seconds. As you said, currently, we are creating all those files at the same time. 

I will add the `sleep` instruction in the `finally` block.